### PR TITLE
Add serializable types and initial toJson() type of call for Tile

### PR DESCRIPTION
**Pull request by kemcbride**

### CODE REVIEW

**Comment by kemcbride** [comment]
```
[2020-06-15T02:20:51Z INFO  team_heist_tactics] All environment variables set
[2020-06-15T02:20:51Z INFO  team_heist_tactics::loadData] Hello world!
[2020-06-15T02:20:51Z INFO  team_heist_tactics::loadData] {"position":{"x":0,"y":0},"squares":[{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_
type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{
"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wal
l":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east
_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"
south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0}]}
[2020-06-15T02:20:51Z INFO  team_heist_tactics::manager] Created game: anotherhandle
[2020-06-15T02:20:51Z INFO  actix_server::builder] Starting 8 workers
[2020-06-15T02:20:51Z INFO  actix_server::builder] Starting "actix-web-service-0.0.0.0:19996" service on 0.0.0.0:19996
[2020-06-15T02:21:21Z INFO  team_heist_tactics::loadData] Hello world!
[2020-06-15T02:21:21Z INFO  team_heist_tactics::loadData] {"position":{"x":0,"y":0},"squares":[{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_
type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{
"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wal
l":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east
_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"
south_wall":0,"west_wall":0,"square_type":0},{"north_wall":0,"east_wall":0,"south_wall":0,"west_wall":0,"square_type":0}]}
[2020-06-15T02:21:21Z INFO  team_heist_tactics::manager] Created game: meme
^[^C[2020-06-15T02:30:52Z INFO  actix_server::builder] SIGINT received, exiting
```


^ output when I added a call to this function from `manager.rs`, to show what it looks like serialized.

File: src/lib.rs [hunk 1]
```diff
@@ -7,3 +7,4 @@ pub mod manager;
 pub mod serializer;
 pub mod types;
 pub mod utils;
+pub mod load_json;
```
Comment (banool) [inline src/lib.rs:10]: What if.......... load_data or load_map or something?

File: src/load_json.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+pub fn load_tiles_from_json() -> HashMap<String, Tile> {
+    // TODO
+    // Ideally takes a path, then returns a hashmap of Tiles
+    let tile_map = HashMap::<String, Tile>::new();
+    tile_map
+}
+
+pub fn serialize_file_test() -> () {
```
Comment (banool) [inline src/load_json.rs:23]: Whack the `#[test]` macro on this and it'll get run when you run `cargo test`.

File: src/load_json.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    // for discovering how to write these
+    let mut my_squares = Vec::<Square>::new();
```
Comment (banool) [inline src/load_json.rs:25]: Prefer `let mut my_squares: Vec<Square> = Vec::new();`

File: src/load_json.rs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    for _i in 0..16 {
```
Comment (banool) [inline src/load_json.rs:26]: Just `_`

File: src/types.rs [hunk 5]
```diff
@@ -150,6 +166,19 @@ impl Internal for Square {
     }
 }
 
+impl From<SerializableSquare> for Square {
+    fn from(item: SerializableSquare) -> Self {
+        let inw = i32::from(item.north_wall);
+        Square {
+            north_wall: WallType::from_i32(inw).unwrap(),
```
Comment (banool) [inline src/types.rs:173]: Make this like the other walls.

File: src/types.rs [hunk 6]
```diff
@@ -360,3 +389,49 @@ impl Internal for InvalidRequest {
...
+        for square in item.squares {
+            serializable_squares.push(SerializableSquare::from(square));
+        }
+        SerializableTile {
+            squares: serializable_squares,
+            position: item.position,
+        }
+    }
+}
+// ---
```
Comment (banool) [inline src/types.rs:437]: killlllll

File: src/lib.rs (hunk 1 above)
Comment (kemcbride) [reply src/lib.rs:10]: load map? (y)

File: src/types.rs (hunk 6 above)
Comment (kemcbride) [reply src/types.rs:437]: it's my _very_ thoughtful comment section ended. but i don't care particularly strongly about keeping the sectioning or not /shrug.keeping for now complain again if you really want it

File: src/types.rs (hunk 5 above)
Comment (kemcbride) [reply src/types.rs:173]: 😨  aye aye

**Comment by kemcbride** [comment]
DIE kill kill die this pull request

I will be creating a new one, since I messed up all my branches. It has... ideally! All the fixes you could possibly want!

**Pull request closed without merging**